src/Microsoft.Extensions.Configuration.UserSecrets/ConfigurationExtensions.cs

```diff
@@ -76,15 +24,15 @@ public static IConfigurationBuilder AddUserSecrets(this IConfigurationBuilder co
                 throw new ArgumentNullException(nameof(configuration));
             }
 
-            var projectPath = configuration.GetProjectPath();
-            if (string.IsNullOrEmpty(projectPath))
+            var fileProvider = configuration.GetFileProvider();
+            if (fileProvider == null)
             {
                 throw new InvalidOperationException(Resources.FormatError_MissingBasePath(
-                        projectPath,
                         typeof(IConfigurationBuilder).Name,
-                        ProjectPathKey));
+                        "FileProvider"));
             }
-            return AddSecretsFile(configuration, PathHelper.GetSecretsPath(projectPath));
+
+            return AddSecretsFile(configuration, PathHelper.GetSecretsPath(fileProvider.GetFileInfo("/").PhysicalPath));
```

Comment:
IMO it would be cleaner if you wrote `GetSecretsPath()` in terms of file providers and the added an overload that took a path for the tool, but I don't really care that much.